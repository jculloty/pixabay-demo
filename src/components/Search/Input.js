import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class SearchInput extends PureComponent {
  static propTypes = {
    value: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleEnter: PropTypes.func.isRequired,
  };
  domElement = React.createRef();

  componentDidMount() {
    if (this.textInput) {
      this.textInput.focus();
    }
  }

  setTextInputRef = (element) => {
    this.textInput = element;
  };

  handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.props.handleEnter();
    }
  }

  handleChange = (event) => {
    this.props.handleChange(event.target.value);
  }

  render() {
    return (
      <input type="text"
        placeholder="Search images, vectors and photos"
        value={this.props.value}
        onChange={this.handleChange}
        ref={this.setTextInputRef}
        onKeyPress={this.handleKeyPress}
        className="form-control"
      />
    );
  }
}

export default SearchInput;
