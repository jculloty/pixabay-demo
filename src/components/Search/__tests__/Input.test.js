import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'test-utils/setup-enzyme';
import Input from '../Input';

const buildComponent = function(props = {}) {
  props = {
    ...{
      value: 'testing',
      handleChange: () => {},
      handleEnter: () => {},
    },
    ...props,
  };

  return (<Input {...props} />);
};

describe('Search: Input', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('focuses when mounted', () => {
    expect(true).toBe(false);
  });

  it('Typing triggeres the correct handler', () => {
    const mockChange = jest.fn();
    const wrapper = shallow(buildComponent({ handleChange: mockChange }));
    const expectedValue = 'Testing...';

    wrapper.find('input').simulate('change', { target: { value: expectedValue } });

    expect(mockChange).toHaveBeenCalledTimes(1);
    expect(mockChange).toHaveBeenCalledWith(expectedValue);
  });

  it('Pressing enter triggeres the correct handler', () => {
    const mockChange = jest.fn();
    const wrapper = shallow(buildComponent({ handleEnter: mockChange }));

    wrapper.find('input').simulate('keypress', { key: 'Enter' });
    expect(mockChange).toHaveBeenCalledTimes(1);
  });
});
