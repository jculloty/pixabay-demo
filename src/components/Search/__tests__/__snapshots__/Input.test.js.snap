// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: Input renders correctly 1`] = `
<input
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Search images, vectors and photos"
  type="text"
  value="testing"
/>
`;
